projects/neural_correlations/EXP1-Random/NeuCorr_Exp1.ipynb

```diff
@@ -173,79 +108,114 @@
-    "            for cell in tm.getActiveCells():\n",
-    "                spikeTrains[cell, ts] = 1            \n",
+    "            tm.compute(allSequences[seqIndices[s]][symbol], learn=True)\n",
     "            \n",
+    "            for cell in tm.getActiveCells():\n",
+    "                spikeTrains[cell, ts] = 1\n",
+    "                numSpikes += 1\n",
+    "                spikeCount[ts] += 1\n",
+    "                \n",
     "            # Obtain active columns:\n",
     "            activeColumnsIndices = [tm.columnForCell(i) for i in tm.getActiveCells()]\n",
     "            currentColumns = [1 if i in activeColumnsIndices else 0 for i in range(tm.numberOfColumns())]\n",
     "            for col in np.nonzero(currentColumns)[0]:\n",
-    "                columnUsage[col] += 1            \n",
-    "            \n",
+    "                columnUsage[col] += 1\n",
+    "                \n",
     "            if ts > 0 and ts % int(totalTS * 0.1) == 0:\n",
-    "                print \"++ Analyzing correlations (cells at random) ++\"                \n",
+    "                numSpikesX.append(ts)\n",
+    "                numSpikesY.append(numSpikes)            \n",
+    "                numSpikes = 0\n",
+    "                \n",
+    "                #print \"++ Analyzing correlations (cells at random) ++\"                \n",
     "                subSpikeTrains = subSample(spikeTrains, 1000, tm.numberOfCells(), ts)\n",
     "                (corrMatrix, numNegPCC) = computePWCorrelations(subSpikeTrains, removeAutoCorr=True)\n",
-    "                negPCCX_cells.append(s)\n",
+    "                negPCCX_cells.append(ts)\n",
     "                negPCCY_cells.append(numNegPCC)                \n",
-    "                print \"++ Generating histogram ++\"\n",
-    "                bins = 100\n",
+    "                bins = 300\n",
     "                plt.hist(corrMatrix.ravel(), bins, alpha=0.5)                \n",
     "                # Set range for plot appropriately!\n",
     "                plt.xlim(-0.05,0.1)\n",
     "                plt.xlabel(\"PCC\")\n",
     "                plt.ylabel(\"Frequency\")\n",
     "                plt.savefig(\"cellsHist_\" + str(ts))\n",
     "                plt.close()\n",
-    "                print \"++ Computing entropy ++\"\n",
     "                entropyX.append(ts)\n",
     "                entropyY.append(computeEntropy(subSpikeTrains))                \n",
-    "                \n",
-    "                print \"++ Analyzing correlations (whole columns) ++\"\n",
+    "\n",
+    "                #print \"++ Analyzing correlations (whole columns) ++\"\n",
     "                ### First the LARGE subsample of columns:\n",
     "                subSpikeTrains = subSampleWholeColumn(spikeTrains, colIndicesLarge, tm.getCellsPerColumn(), ts)\n",
     "                (corrMatrix, numNegPCC) = computePWCorrelations(subSpikeTrains, removeAutoCorr=True)\n",
```

Comment:
in that line I should have invoked this function: "computePWCorrelationsWithinCol". My bad